### PR TITLE
TIP-500: Dispatch events during installation process

### PR DESCRIPTION
**Pull request by fitn**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**

The goal is to provide a way to add custom needed fixtures or command in an extension or a project without overriding the installer.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo 
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/InstallerBundle/Command/DatabaseCommand.php [hunk 1]
```diff
@@ -144,6 +153,7 @@ protected function createNotMappedTables(OutputInterface $output)
      */
     protected function loadFixturesStep(InputInterface $input, OutputInterface $output)
     {
+
```
Comment (anaelChardan) [inline src/Pim/Bundle/InstallerBundle/Command/DatabaseCommand.php]: ?

File: src/Pim/Bundle/InstallerBundle/Command/DatabaseCommand.php [hunk 2]
```diff
@@ -102,11 +105,17 @@ protected function execute(InputInterface $input, OutputInterface $output)
                 ['--force' => true, '--no-interaction' => true]
             );
 
+        $this->getEventDispatcher()->dispatch(InstallerEvents::POST_DB_CREATE);
```
Comment (anaelChardan) [inline src/Pim/Bundle/InstallerBundle/Command/DatabaseCommand.php]: Question: no PRE_DB_CREATE before ?

Comment (fitn) [inline src/Pim/Bundle/InstallerBundle/Command/DatabaseCommand.php]: For now I don't see any use case for that event :)

Comment (anaelChardan) [inline src/Pim/Bundle/InstallerBundle/Command/DatabaseCommand.php]: Me neither, it was just a question to be sure to not forget something :-)

File: src/Pim/Bundle/InstallerBundle/Event/InstallerEvents.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+ * Events dispached during installation process
+ *
+ * @author    Romain Monceau <romain@akeneo.com>
+ * @copyright 2016 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class InstallerEvents
+{
+    /**
+     * This event is dispatched after having install the databases (ORM + MongoDB)
```
Comment (phaseinducer) [inline src/Pim/Bundle/InstallerBundle/Event/InstallerEvents.php]: installed

File: src/Pim/Bundle/InstallerBundle/Event/InstallerEvents.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+     *
+     * You can use it to create new tables that are not managed with doctrine.
+     *
+     * @const string
```
Comment (phaseinducer) [inline src/Pim/Bundle/InstallerBundle/Event/InstallerEvents.php]: No need to document the types on constants

File: src/Pim/Bundle/InstallerBundle/Event/InstallerEvents.php [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+     */
+    const POST_ASSETS_DUMP = 'pim_installer.post_assets_dump';
+
+    /**
+     * This event is dispatched before each fixture load.
+     *
+     * The event listener receives an
+     * Symfony\Component\EventDispatcher\GenericEvent instance with the job profile code.
```
Comment (phaseinducer) [inline src/Pim/Bundle/InstallerBundle/Event/InstallerEvents.php]: with -> containing ?

**Comment by nidup** [comment]
Thx @fitn :tada:

**Pull request merged**